src/app/new-auth/login/login.component.ts

```diff
@@ -1,23 +1,121 @@
-import { Component, OnInit } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { NgForm } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
 import { AuthValidationService } from 'app/core/auth-validation.service';
+import { AuthService } from 'app/core/auth.service';
 
 @Component({
   selector: 'clark-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.scss']
+  styleUrls: ['./login.component.scss'],
+  animations: [
+    trigger(
+      'inOutAnimation',
+      [
+        transition(
+          ':enter',
+          [
+            style({ height: 0, opacity: 0 }),
+            animate('.5s ease-out',
+                    style({ height: 85, opacity: 1 }))
+          ]
+        ),
+        transition(
+              ':leave',
+              [
+                style({ height: 85, opacity: 1 }),
+                animate('.5s ease-in',
+                        style({ height: 0, opacity: 0 }))
+              ]
+            )
+          ]
+        )
+      ]
 })
 export class LoginComponent implements OnInit{
 
-  loginFailure = false;
-  constructor(public authValidation: AuthValidationService) { }
+  loginFailure: Boolean = false;
+  isNameLogin = false;
+  authInfo: {username: string, password: string};
+  errorMsg = '';
+  bannerMsg = '';
+  attempts = 0;
+  @ViewChild('username')
+  username;
+  @ViewChild('password')
+  password;
+  redirectUrl;
+  constructor(
+    private authValidation: AuthValidationService,
+    private route: ActivatedRoute,
+    private auth: AuthService,
+    private router: Router
+    ) {
+      this.route.parent.data.subscribe(() => {
+        if (route.snapshot.queryParams.redirectUrl) {
+          this.redirectUrl = decodeURIComponent(route.snapshot.queryParams.redirectUrl);
+        }
+      });
```

Comment:
You'll also want to check if there is an err message for single sign on to display to the user.

Reply:
^

Reply:
Ok so I added displaying the error message on the front end, but, when i try to test it with an account that isn't registered in the database, it doesn't redirect correctly. it's almost like when an error happens it doesn't hit the catch in handleGoogleRedirect(). it's handled somewhere else.

Reply:
Check to see if there is an errorParser or errorHandler in a catch somewhere. Some of them are setup to throw out a bunch of errors